provider/openstack/provider.go

```diff
@@ -1017,20 +1017,40 @@ func (e *Environ) StartInstance(args environs.StartInstanceParams) (*environs.St
 		networks = append(networks, nova.ServerNetworks{NetworkId: networkId})
 	}
 
-	var apiPort int
-	if args.InstanceConfig.Controller != nil {
-		apiPort = args.InstanceConfig.Controller.Config.APIPort()
-	} else {
-		// All ports are the same so pick the first.
-		apiPort = args.InstanceConfig.APIInfo.Ports()[0]
-	}
-	groupNames, err := e.firewaller.SetUpGroups(args.ControllerUUID, args.InstanceConfig.MachineId, apiPort)
-	if err != nil {
-		return nil, errors.Annotate(err, "cannot set up groups")
+	// For BUG 1680787: openstack: add support for neutron networks where port security is disabled.
+	// If PortSecurityEnabled is false, don't create security groups, instance boot will fail.
+	createSecurityGroups := true
+	// there will be 1 or no networks in []networks based finding network code
+	if len(networks) == 1 && e.supportsNeutron() {
+		client := e.neutron()
+		net, err := client.GetNetworkV2(networks[0].NetworkId)
+		if err != nil {
+			return nil, err
+		}
+		if net.PortSecurityEnabled != nil {
+			createSecurityGroups = *net.PortSecurityEnabled
+		}
 	}
-	novaGroupNames := make([]nova.SecurityGroupName, len(groupNames))
-	for i, name := range groupNames {
-		novaGroupNames[i].Name = name
+
+	var novaGroupNames = []nova.SecurityGroupName{}
+	if createSecurityGroups {
+		var apiPort int
+		if args.InstanceConfig.Controller != nil {
+			apiPort = args.InstanceConfig.Controller.Config.APIPort()
+		} else {
+			// All ports are the same so pick the first.
+			apiPort = args.InstanceConfig.APIInfo.Ports()[0]
+		}
+		groupNames, err := e.firewaller.SetUpGroups(args.ControllerUUID, args.InstanceConfig.MachineId, apiPort)
+		if err != nil {
+			return nil, errors.Annotate(err, "cannot set up groups")
+		}
+		novaGroupNames = make([]nova.SecurityGroupName, len(groupNames))
+		for i, name := range groupNames {
+			novaGroupNames[i].Name = name
+		}
+	} else {
+		logger.Infof("network %q has port_security_enabled set to false. Not using security groups.", networks[0].NetworkId)
```

Comment:
if networks is 0, createSecurityGroups will still be true, still moving the log msg.  

but that does raise a question to me, of why do we try to boot if there are no networks, since it will fail unless there is only 1 network of any type available. filing a bug to look at improved messaging if openstack provider bootstrap etc fails due to lack of specified network.